sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_fl_scatter_gather_node.py

```diff
@@ -0,0 +1,115 @@
+            try to replace it.
+
+    The process of trimming down inputs from the silo steps to a single input for the aggregate step has some
+    caveats:
+        - Only silo outputs of type mltable and uri_folder are supported.
+        - Both the above output types become an mltable which mounts all the silo step outputs as sub-folders.
+
+    The expected use case of this node is shown in the following code snippet:
+        ```
+        @pipeline()
+        def fl_pipeline():
+            fl_node = fl_scatter_gather(**many_inputs)
+            return {"pipeline_output" : fl_node.outputs["aggregated_model"]}
+
+        submitted_pipeline_job = my_client.jobs.create_or_update(fl_pipeline(), experiment_name="example_fl_pipeline")
+        ```
+
+    param silo_component: A pipeline step that will be run multiple
+        times across different silos, as specified by the silo_configs input. In a typical
+        horizontal federated learning context, this step is what will perform model
+        training using siloed subsets of data. Can be either a sub-pipelineJob, or a command component
+    type silo_component: Union[PipelineJob, CommandComponent]
+    param aggregation_component: A pipeline step which receives inputs from the myriad executed silo components,
+        and does something with them. In a typical horizontal federated learning context, this component
+        will merge the models that were independently trained on each silo's data in a single model.
+        Can be either a sub-pipelineJob, or a command component
+    type aggregation_component: Union[PipelineJob, CommandComponent]
+        param silo_configs: A list of FederatedLearningSilo objects, which contain the necessary data
+        to reconfigure components to run on specific computes and datastores, while also targeting
+        specific inputs located on the aforementioned datastores.
+    type silo_configs: List[FederatedLearningSilo]
+    param aggregation_compute: The name of the compute that the aggregation component will use.
+    type aggregation_compute: string
+    param aggregation_datastore: The name of the datastore that the aggregation component will use.
+    type aggregation_datastore: string
+    param shared_silo_kwargs: A dictionary of string keywords to component inputs. This dictionary is treated
+        like kwargs, and is injected into ALL executed silo components.
+    type shared_silo_kwargs: Dict
+    param aggregation_kwargs: A dictionary of string keywords to component inputs. This dictionary is treated
+        like kwargs, and is injected into ALL executed aggregation components.
+    type aggregation_kwargs: Dict
+    param max_iterations: The maximum number of scatter gather iterations that should be performed.
+    type max_iterations: int
+    """
+    # Private kwargs:
+    # _create_default_mappings_if_needed: if true, then try to automatically create i/o mappings if they're unset.
+
```

Comment:
Since this is in a somewhat odd place of depending on a package that was intentionally not added a dependency, consider adding a runtime check that errors and provides a solution for users who use this without mldesigner installed.

Maybe something similar to this pseudocode.

```python
try:
   import mldesigner
   MLDESIGNER_IS_INSTALLED=True
except ImportError:
   MLDESIGNER_IS_INSTALLED=False


def fl_scatter_gather(...):
   if not MLDESIGNER_IS_INSTALLED:
         raise Error(...)
```

Reply:
Added a helper function that checks the import, used it, and added a quick sanity test for it.